pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -325,3 +377,7 @@ func isProviderConfigurable(provider string) bool {
 
 	return false
 }
+
+func isNewSecretValue(value string) bool {
+	return value != setting.RedactedPassword
```

Comment:
> Instead of relying on a fixed string to indicate that the password hasn't been changed isn't it better to introduce a bool in the http clients to reflect whether the password is new? 

I think both could work, I implemented this, because I would prefer not to add a field to the API contract/model that is specific to the UI.